bindings/cpp/logger.cpp

```diff
@@ -156,32 +154,35 @@ void dnet_node_set_trace_id(dnet_logger *logger, uint64_t trace_id, int tracebit
 			attributes.insert(std::make_pair(std::string("backend_id"), blackhole::log::attribute_t(backend_id)));
 		}
 
-		new (local_attributes) scoped_attributes_t(*logger, std::move(attributes));
+		scoped_attributes[scoped_count] = new scoped_attributes_t(*logger, std::move(attributes));
```

Comment:
If you remove `scoped_buffer`, there is no profit to use `blackhole::scoped_attributes_t *scoped_attributes[DNET_SCOPED_LIMIT]` and it is better to replace it by `std::vector<blackhole::scoped_attributes_t >`.